### ISSUE TITLE
Adapt RSSM to Accept MLP Observation Data in the Encoder

### ISSUE THREAD

**Issue by waterkimchi** [uci-f1tenth/uci_f1tenth_workshop#16]
## Description:

Modify the `networks.py` file to replace the existing Convolutional Neural Network (CNN) architecture with a Multi-Layer Perceptron (MLP) to process the 1D LiDAR data.  CNNs are designed for image data (2D grids), while MLPs are more appropriate for 1D vector data like LiDAR scans.  This change is essential for effectively processing the LiDAR data within the DreamerV3 agent.

**We currently do not have deep knowledge of how DreamerV3 works, this issue may be solvable as easy as changing a hyper parameter for the run script, or as hard as having to change an entire CNN to MLP implementation.**

## Current Issue:

The current `networks.py` file or other files that handle the RSSM algorithm likely uses a CNN architecture(or not), which is not suitable for processing the 1D LiDAR data.  CNNs are designed to exploit spatial hierarchies in 2D data (like images), a property that 1D LiDAR scans do not possess.  Using a CNN for LiDAR data would be inefficient and likely lead to suboptimal performance.

## Expected Behavior:

1. **MLP Implementation:**
   - Create a new class (or modify an existing one) in `networks.py` to define an MLP network.  This MLP should:
     - Take the processed LiDAR data (after normalization and any other preprocessing in `racerenv.py`) as input.  The input dimension should match the size of the processed LiDAR vector.
     - Consist of multiple fully connected (linear) layers.
     - Include appropriate activation functions (e.g., ReLU, Tanh) between the linear layers.  The choice of activation function and the number of layers/neurons will likely require experimentation and tuning.
     - Output the features that will be used by the DreamerV3 agent (e.g., for the RSSM or actor-critic networks).

2. **Network Integration:**
   - Replace the existing CNN architecture in the relevant parts of the DreamerV3 model (e.g., within the RSSM, actor, or critic networks) with the newly created MLP.  This will likely involve changing how the input data is processed and passed through the network.

3. **Configuration:**
   - Ensure that the network configuration (number of layers, number of neurons per layer, activation functions, etc.) can be easily configured, ideally through the `configs.py` file.  Avoid hardcoding these parameters directly in `networks.py`.

## Code Implementation (Pseudo-code):

```python
# networks.py
class LidarMLP(nn.Module):  # Or modify an existing class
    def __init__(self, input_dim, hidden_dim1=64, hidden_dim2=64, ...):  # Configurable!
        super().__init__()
        self.mlp = nn.Sequential(
            nn.Linear(input_dim, hidden_dim1),
            nn.ReLU(),  # Or other activation
            nn.Linear(hidden_dim1, hidden_dim2),
            nn.ReLU(), # Or other activation
            # ... more layers as needed ...
            nn.Linear(hidden_dimN, output_dim), # output dim depends on where the MLP output is used.
        )

    def forward(self, x):
        return self.mlp(x)


# dream.py (or wherever the network is instantiated)
input_dim = ... # Should match the output dimension after the data processing in racerenv.py
lidar_mlp = LidarMLP(input_dim, hidden_dim1=config.hidden_dim1, hidden_dim2=config.hidden_dim2, ...) # Use config values

# ... (Rest of the DreamerV3 model definition, replacing CNN with the MLP)

# configs.py
config.hidden_dim1 = 64 # Example
config.hidden_dim2 = 64 # Example
# ... other config parameters
